spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ScopePassingSpanSubscriber.java

```diff
@@ -59,25 +60,25 @@
 
 	@Override public void onSubscribe(Subscription subscription) {
 		this.s = subscription;
-		try (Tracer.SpanInScope inScope = this.tracer.withSpanInScope(this.span)) {
+		try (CurrentTraceContext.Scope inScope = this.tracing.currentTraceContext().maybeScope(this.spanTraceContext)) {
```

Comment:
if this change is about efficiency, probably you can inject the `CurrentTraceContext` object and use it directly. I'm not sure any of this code needs the Span object.

cc also @dmexe as there's a beautiful, if unrelated, change going on in reactor as well https://github.com/reactor/reactor-netty/pull/622#issuecomment-468353910